core/capabilities/streams/trigger_load_test.go

```diff
@@ -0,0 +1,224 @@
+package streams_test
+
+import (
+	"crypto/ed25519"
+	"crypto/rand"
+	"math/big"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	ocrTypes "github.com/smartcontractkit/libocr/offchainreporting2plus/types"
+
+	ragetypes "github.com/smartcontractkit/libocr/ragep2p/types"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/datastreams"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/pb"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/triggers"
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/remote"
+	remotetypes "github.com/smartcontractkit/chainlink/v2/core/capabilities/remote/types"
+	"github.com/smartcontractkit/chainlink/v2/core/capabilities/streams"
+	"github.com/smartcontractkit/chainlink/v2/core/internal/testutils"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/chaintype"
+	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/keys/ocr2key"
+	p2ptypes "github.com/smartcontractkit/chainlink/v2/core/services/p2p/types"
+)
+
+const (
+	triggerID          = "streams-trigger@1.0.0"
+	workflowID         = "workflowID1"
+	baseTriggerEventID = "triggerEvent_"
+	basePrice          = 2000000000
+	baseTimestamp      = 1000000000
+)
+
+type feed struct {
+	feedID    [32]byte
+	feedIDStr string
+	reports   []report
+}
+
+type report struct {
+	rawReport  []byte
+	reportCtx  []byte
+	signatures [][]byte
+}
+
+type node struct {
+	peerID p2ptypes.PeerID
+	bundle ocr2key.KeyBundle
+}
+
+// Load test that combines Trigger Subscriber, Streams Trigger Aggregator and Streams Codec.
+// It measures time needed to receive and process trigger events from multiple nodes and produce a local aggregated event.
+// For more meaningful measurements, increase the values of parameters P and T.
+func TestStreamsTrigger_Load(t *testing.T) {
```

Comment:
how do we run and monitor load tests in ci? this won't be run for PR right?

Reply:
The parameters default to very low values (P=2, T=2) so the test takes less than 100ms to run. The "load" part of it won't be exercised in CI but it still tests integration of certain components and validates outputs so I think there is value of having it executed on every PR.

In terms of actual load testing, so far I've only done manual runs with increased params to measure time and profile it. I'm open to suggestions for how to automate it. Maybe we could have a catch-all test that is disabled in CI but runs all load test with high params? (still to be run manually).

Overall I'm skeptical about running load tests in CI. They waste resources and CI env likely doesn't have consistent perf (in fact good CI should vary CPU load on purpose to catch flakes).

Reply:
yes, i'm also skeptical, which is the reason i asked.

it would be ideal to have nightly perf tests in a stable environment that we could monitor for trends.

the test name is misleading given that this test can't be run in an automated way to find load related problem, perhaps `integration` or `sanity` are better than `load`

Reply:
Renamed to simply "TestStreamsTrigger"